### DISCUSSION TITLE
How to run a Porter Bundle from a Kubernetes POD

### DISCUSSION THREAD

**Discussion by srichell** [getporter/porter#1614] [Q&A]
Hi,

I am new to porter and still in the trial mode. I want to Author a Porter Bundle Run it in the Context of a Kubernetes POD (Well, to be more precise, an Argo Workflow, but an Argo Workflow schedules a Set of Kubernetes PODs.).

I have the following Questions:

1. A Typical Workflow Activity is a Kubernetes POD scheduled using Vanilla container image, which on startup will install porter,  get the Porter bundle and will execute a porter command ([  invoke run | invoke monitor | invoke generate-report]). Is there a better way to get this done? Will this even work?
2. Currently, Parameters to the Workflow Activities are passed via env vars defined from Kubernetes Secrets. When I embed my My Activity Logic inside a Porter bundle, will the Env Vars Propagate? Or, is there something I have to do?

**Comment by carolynvs** [comment]
I think you would be interested in the [operator](https://github.com/getporter/operator) that I'm working on for Porter. The operator executes a bundle inside a kubernetes job based on a CRD representing the installation. It's not completed yet, but it sounds like this would work well for you once it's finished. Right now it only executes bundles, it doesn't _build and publish_ bundles, though I'd like to see that too.

Currently, the installation custom resource (here's an [example](https://github.com/getporter/operator/blob/main/config/samples/exec-outputs-add-user.yaml)) is imperative (i.e. you tell it which action to run), I'm still working on desired state, where it reconciles the requested state in the custom resource against the state in Porter's database. I would love help with the operator's design and implementation if you are interested!

If the operator isn't a good fit for you right now, I just updated the documentation for the docker image and manifests that I'm using in the operator to run porter in a pod

https://porter.sh/docker-images/agent/

**Reply by srichell** [reply]
Will explore both options. Thanks
